index.js

```diff
@@ -34,7 +35,22 @@ cmdShim.ifExists = cmdShimIfExists
  * @property {string} extension The file extension for the shim.
  */
 
-const fs = require('mz/fs')
+const {promisify} = require('util')
+
+// fs can be dependency-injected, which we use for testing.
+let fs
+let fsRaw
+setFs(require('fs'))
+function setFs (_fs) {
```

Comment:
why don't you just pass in fs via an optional `opts.fs`?

Reply:
I was avoiding changes to the API.  But if you're ok adding the new option, that works too.

Reply:
yeah, that seems like a widespread convention. Let's add this via an option